src/dsymbol/conversion/first.d

```diff
@@ -1106,7 +1113,7 @@ private:
 	}
 
 	SemanticSymbol* allocateSemanticSymbol(string name, CompletionKind kind,
-		istring symbolFile, size_t location = 0)
+		istring symbolFile, size_t location = 0, IdType protection = tok!"public")
```

Comment:
I think a default value is a bad idea. I would revert this definition, make it deprecated and add another overload with non-optional protection argument. This way we don't get issues with people assuming no protection and also simplify a lot of caller code which currently manually sets acSymbol.protection which isn't covered by the PR yet.

Reply:
A few points to keep in mind:
* I think that this default is a better one than the previous default of `tok!""`, which is much less valid than `public`. `public` is the default for class members, and I'm pretty sure that it's the default for other things as well, but it's not actually written down in the language spec.
* `allocateSemanticSymbol` and `pushSymbol` are both private, so we're not really breaking an API.

I do plan on doing another pass over this code to make more use of this extra argument. In general this module needs a lot of cleanup and documentation.